lib/tapioca/compilers/dsl/active_job.rb

```diff
@@ -0,0 +1,71 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "parlour"
+
+begin
+  require "active_job"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Compilers
+    module Dsl
+      # `Tapioca::Compilers::Dsl::ActiveJob` generates RBI files for subclasses of
+      # [`ActiveJob::Base`](https://api.rubyonrails.org/classes/ActiveJob/Base.html).
+      #
+      # For example, with the following `ActiveJob` subclass:
+      #
+      # ~~~rb
+      # class NotifyUserJob < ActiveJob::Base
+      #   def perform(user)
+      #     # ...
+      #   end
+      # end
+      # ~~~
+      #
+      # this generator will produce the RBI file `notify_user_job.rbi` with the following content:
+      #
+      # ~~~rbi
+      # # notify_user_job.rbi
+      # # typed: true
+      # class NotifyUserJob
+      #   sig { params(user: T.untyped).returns(T.attached_class) }
+      #   def self.perform_later(user); end
+      #
+      #   sig { params(user: T.untyped).returns(T.attached_class) }
+      #   def self.perform_now(user); end
+      # end
+      # ~~~
+      class ActiveJob < Base
+        extend T::Sig
+
+        sig { override.params(root: Parlour::RbiGenerator::Namespace, constant: T.class_of(::ActiveJob::Base)).void }
+        def decorate(root, constant)
+          root.path(constant) do |job|
+            next unless constant.instance_methods(false).include?(:perform)
+
+            method = constant.instance_method(:perform)
+            parameters = compile_method_parameters_to_parlour(method)
+
+            %w[perform_later perform_now].each do |name|
+              create_method(
+                job,
+                name,
+                parameters: parameters,
+                return_type: "T.attached_class",
```

Comment:
Yeah, it was for the runtime side of things.

Basically if you did something like:

```ruby
class MyJob < AJ::Base
  def perform; end
end

class MyOtherJob < MyJob
  def perform; end
  def my_helper; end
end
```

and then you went:

```ruby
MyOtherJob.perform_later.my_helper
```

but you had only generated RBI for `MyJob`, then it would fail. That being said, it's kind of a dumb use case, because it would only happen if you were dynamically creating the class instead of using tapioca as designed. So I'll go with `constant.name`.